contracts/controller/src/healthfactor.rs

```diff
@@ -116,18 +121,24 @@ mod tests {
         );
 
         let mut prices: Vec<Price> = Vec::new();
-        prices.push(Price {
-            ticker_id: ticker_id_near,
-            value: U128(20000),
-            volatility: U128(80),
-            fraction_digits: 4,
-        });
-        prices.push(Price {
-            ticker_id: ticker_id_eth,
-            value: U128(20000),
-            volatility: U128(100),
-            fraction_digits: 4,
-        });
+
+        if near_price != 0 {
```

Comment:
@dmytro-shash you cannot change existing tests (until they are broken). If you need test without zero price you may create separate fixuture for that

Reply:
I meant to add the price of eth/near if they needed ( price are not  = 0 )

Reply:
there is few problems 

1) you may want to construct `prices` like this 
```
let prices: Vec<Price> = vec![ Price {..}, ...];
```
in this case you stick to RAII and immutability principles

2) it's not clear for the user of `init_price_volatility` why price where not added (someone may want add zero prices for some cases)

3) `init_price_volatility` has now actually 4 version of behavior (near_price == 0, near_price != 0, eth_price == 0, eth_price != 0) which in case of fixture is not that bad but in case of regular function requires 4x more tests

so if zero prices are not a problem or if you can implement separate function without prices and with it shuold be preferable

Reply:
solved 🙏🏻